tests/SprykerEcoTest/Zed/AmazonQuicksight/Business/Facade/CreateQuicksightUsersForUserTransfersTest.php

```diff
@@ -0,0 +1,199 @@
+
+use Aws\QuickSight\Exception\QuickSightException;
+use Aws\Result;
+use Aws\ResultInterface;
+use Codeception\Test\Unit;
+use Generated\Shared\Transfer\UserCollectionResponseTransfer;
+use Generated\Shared\Transfer\UserTransfer;
+use SprykerEco\Zed\AmazonQuicksight\AmazonQuicksightDependencyProvider;
+use SprykerEco\Zed\AmazonQuicksight\Dependency\External\AmazonQuicksightToAwsQuicksightClientInterface;
+use SprykerEcoTest\Zed\AmazonQuicksight\AmazonQuicksightBusinessTester;
+
+class CreateQuicksightUsersForUserTransfersTest extends Unit
+{
+    /**
+     * @uses \SprykerEco\Zed\AmazonQuicksight\AmazonQuicksightConfig::QUICKSIGHT_USER_ROLE_READER
+     *
+     * @var string
+     */
+    protected const QUICKSIGHT_USER_ROLE_READER = 'READER';
+
+    /**
+     * @var string
+     */
+    protected const ERROR_MESSAGE_QUICKSIGHT_USER_REGISTER_FAILURE = 'An internal failure occurred.';
+
+    /**
+     * @uses \SprykerEco\Zed\AmazonQuicksight\Business\ApiClient\AmazonQuicksightApiClient::ERROR_MESSAGE_USER_REGISTRATION_FAILED
+     *
+     * @var string
+     */
+    protected const ERROR_MESSAGE_USER_REGISTRATION_FAILED = 'Failed to register Quicksight user.';
+
+    /**
+     * @var \SprykerEcoTest\Zed\AmazonQuicksight\AmazonQuicksightBusinessTester
+     */
+    protected AmazonQuicksightBusinessTester $tester;
+
+    /**
+     * @return void
+     */
+    public function testShouldPersistQuicksightUserWhenUserSuccessfullyRegisteredInQuicksight(): void
+    {
+        // Arrange
+        $userTransfer = $this->tester->haveUserWithNotPersistedQuicksightUserRole(static::QUICKSIGHT_USER_ROLE_READER);
+
+        $awsQuicksightClientMock = $this->getAwsQuicksightClientMock();
+        $awsQuicksightClientMock->method('registerUser')
+            ->willReturn($this->createRegisterUserSuccessfulResponse($userTransfer));
+        $this->tester->setDependency(
+            AmazonQuicksightDependencyProvider::AWS_QUICKSIGHT_CLIENT,
+            $awsQuicksightClientMock,
+        );
+
+        $userCollectionResponseTransfer = (new UserCollectionResponseTransfer())->addUser($userTransfer);
+
+        // Act
+        $userCollectionResponseTransfer = $this->tester->getFacade()
+            ->createQuicksightUsersForUserTransfers($userCollectionResponseTransfer);
+
+        // Assert
+        $this->assertCount(0, $userCollectionResponseTransfer->getErrors());
+        $this->assertCount(1, $userCollectionResponseTransfer->getUsers());
+
+        /** @var \Generated\Shared\Transfer\UserTransfer $userTransfer */
+        $userTransfer = $userCollectionResponseTransfer->getUsers()->getIterator()->current();
+        $this->assertNotNull($userTransfer->getQuicksightUser());
+        $this->assertNotNull($userTransfer->getQuicksightUserOrFail()->getIdQuicksightUser());
+        $this->assertNotNull($userTransfer->getQuicksightUserOrFail()->getArn());
+        $this->assertNotNull($userTransfer->getQuicksightUserOrFail()->getPrincipalId());
```

Comment:
The test name tells that we are checking that the user persisted but we are actually checking that some data is returned. So we need to have one more test checking the DB